src/components/presentational/NewsDetail.js

```diff
@@ -8,6 +8,61 @@ import url from 'url';
 import convertNumberToTimeAgo from '../../helpers/convertNumberToTimeAgo';
 
 export default class NewsDetail extends Component {
+  // Constructor method to initiate the "bookmarked" state
+  constructor(props) {
+    super(props);
+    this.state = {
+      bookmarked: false,
+    }
+  }
+  componentDidMount() {
+      this.hydrateStateWithLocalStorage();
+   }
+  //Method to hydrate state component to localStorage
+  hydrateStateWithLocalStorage() {
+    //Handle case when user is not logged in
+    let userId = "guest"
+    if(this.props.me) {
+      userId = this.props.me.id;
+    }
+    if (localStorage.hasOwnProperty(userId)) {
+      // get the key's value from localStorage
+      const userBookmarks = JSON.parse(localStorage.getItem(userId)) || [];
+      //Look if the post has been bookmarked and set the state accordingly
+      const indexItemBookmarked = userBookmarks.findIndex((postItem,i) => postItem.id == this.props.id);
+      indexItemBookmarked !== -1 ? this.setState({ bookmarked: true }) : this.setState({ bookmarked: false });
```

Comment:
`this.setState({ bookmarked: indexItemBookmarked !== -1  })` would have produced the same